conformance-tests/Makefile

```diff
@@ -0,0 +1,52 @@
+GATEWAY_CLASS = nginx
+SUPPORTED_FEATURES = Gateway,HTTPRoute
+KIND_KUBE_CONFIG_FOLDER = $${HOME}/.kube/kind
+TAG = latest
+PREFIX = conformance-test-runner
+.DEFAULT_GOAL := help
```

Comment:
I think we should make so that by default, it runs the image built by a developer. So that as I developer, I can conformance tests locally easily.

One possible apporoach:
- Add NGK_PREFIX and NKG_TAG variables. Set to NKG_PREFIX=nginx-kubernetes-gateway and NKG_TAG=edge by default. Make sure those are documented in the README, so that people can customize it to run an image from a different repo.
- Add a target `prepare-nkg` that builds and loads NKG  (using those variables above) image into the kind cluster.  Add it as a step in the README.
- Add IMAGE_PULL_POLICY with the default Never. Make sure that the installation command modify the policy for both the NKG static mode deployment and provisioner deployment. This is necessary so that Kind doesn't try to pull a local image from the Docker repo. Also, better to document that variable in the README too.

Reply:
@pleshakov not sure about how to approach the last point here, are we expecting Makefile to update `/conformance/provisioner/provisioner.yaml` and `deploy/manifests/deployment.yaml`  with `image` and `imagePullPolicy` without having template files for each of those? can you point me to a place where its done?

Reply:
ok. I see the problem now. The approach you went with looks reasonable to me